subsys/bluetooth/mesh/light_ctl_srv.c

```diff
@@ -0,0 +1,618 @@
+		.target.delta_uv = status.target.delta_uv,
+		.remaining_time = status.remaining_time,
+	};
+	struct bt_mesh_lvl_status lvl_status = {
+		.current = temp_to_lvl(&srv->temp_srv, status.current.temp),
+		.target = temp_to_lvl(&srv->temp_srv, status.target.temp),
+		.remaining_time = status.remaining_time,
+	};
+	struct bt_mesh_lightness_status lightness_status = {
+		.current = status.current.light,
+		.target = status.target.light,
+		.remaining_time = status.remaining_time,
+	};
+
+	light_ctl_pub(srv, &status);
+	(void)bt_mesh_light_ctl_temp_srv_pub(&srv->temp_srv, NULL,
+					     &temp_status);
+	(void)bt_mesh_lvl_srv_pub(&srv->temp_srv.lvl, NULL, &lvl_status);
+	(void)bt_mesh_lightness_srv_pub(&srv->lightness_srv, NULL,
+					&lightness_status);
+
+respond:
+	if (ack) {
+		ctl_rsp(model, ctx, &status);
+	}
+}
+
+static void handle_ctl_get(struct bt_mesh_model *model,
+			   struct bt_mesh_msg_ctx *ctx,
+			   struct net_buf_simple *buf)
+{
+	struct bt_mesh_light_ctl_srv *ctl_srv = model->user_data;
+	struct bt_mesh_light_ctl_gen_cb_status status;
+
+	ctl_srv->handlers->get(ctl_srv, NULL, &status);
+	ctl_rsp(model, ctx, &status);
+}
+
+static void handle_ctl_set(struct bt_mesh_model *model,
+			   struct bt_mesh_msg_ctx *ctx,
+			   struct net_buf_simple *buf)
+{
+	ctl_set(model, ctx, buf, true);
+}
+
+static void handle_ctl_set_unack(struct bt_mesh_model *model,
+				 struct bt_mesh_msg_ctx *ctx,
+				 struct net_buf_simple *buf)
+{
+	ctl_set(model, ctx, buf, false);
+}
+
+static void range_encode_status(struct net_buf_simple *buf,
+				struct bt_mesh_light_ctl_srv *srv,
+				enum bt_mesh_model_status status)
+{
+	bt_mesh_model_msg_init(buf, BT_MESH_LIGHT_CTL_TEMP_RANGE_STATUS);
+	net_buf_simple_add_u8(buf, status);
+	net_buf_simple_add_le16(buf, srv->temp_srv.temp_range.min);
+	net_buf_simple_add_le16(buf, srv->temp_srv.temp_range.max);
+}
+
+static void temp_range_rsp(struct bt_mesh_model *model,
+			   struct bt_mesh_msg_ctx *rx_ctx,
+			   enum bt_mesh_model_status status)
+{
+	struct bt_mesh_light_ctl_srv *srv = model->user_data;
+
+	BT_MESH_MODEL_BUF_DEFINE(msg, BT_MESH_LIGHT_CTL_TEMP_RANGE_STATUS,
+				 BT_MESH_LIGHT_CTL_MSG_LEN_TEMP_RANGE_STATUS);
+	range_encode_status(&msg, srv, status);
+	(void)bt_mesh_model_send(model, rx_ctx, &msg, NULL, NULL);
+}
+
+static void temp_range_set(struct bt_mesh_model *model,
+			   struct bt_mesh_msg_ctx *ctx,
+			   struct net_buf_simple *buf, bool ack)
+{
+	if (buf->len != BT_MESH_LIGHT_CTL_MSG_LEN_TEMP_RANGE_SET) {
+		return;
+	}
+
+	struct bt_mesh_light_ctl_srv *srv = model->user_data;
+	enum bt_mesh_model_status status;
+	uint16_t new_min = net_buf_simple_pull_le16(buf);
+	uint16_t new_max = net_buf_simple_pull_le16(buf);
+
+	if (new_min < BT_MESH_LIGHT_CTL_TEMP_RANGE_MIN) {
+		status = BT_MESH_MODEL_ERROR_INVALID_RANGE_MIN;
+		goto respond;
+	} else if (new_max > BT_MESH_LIGHT_CTL_TEMP_RANGE_MAX) {
+		status = BT_MESH_MODEL_ERROR_INVALID_RANGE_MAX;
+		goto respond;
+	}
```

Comment:
`else if (new_min > new_max)` ?

Reply:
Hmm... I agree. But what action to take? No corresponding error code that fits for this issue. Maybe:
`	} else if (new_max < new_min) {
                new_min = new_max;
	}` ???

Reply:
Re: I discovered that my solution is not a great idea, and at the same time I discovered a potential bug-case. If MIN_RANGE = MAX_RANGE we will get a division be zero error from the conversion function for temp_to_lvl, so either we must prevent MIN_RANGE = MAX_RANGE, or handle the division by zero case in temp_to_lvl in some manner.

Reply:
Yeah, it doesn't really matter which error code you use in this case, but it shouldn't be accepted.